### PR TITLE
Rename DataProxy to ESProductResolver

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

The EventSetup's DataProxy was poorly named (it is not a proxy). The name was changed to match the equivalent naming used in the Run/LuminosityBlock/Event system. All related names were also changed.

#### PR validation:

Code compiles. This is strictly meant to be a refactoring of names so no functionality should be affected.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

**Comment by makortel** [comment]
+core

Unit test fails occasionally also in IBs, workflow 12434.0 shows #41200 .

**Comment by Dr15Jones** [comment]
ping @cms-sw/alca-l2 @cms-sw/db-l2 @cms-sw/geometry-l2 @cms-sw/simulation-l2

File: CondCore/ESSources/interface/ProductResolver.h [hunk 1]
```diff
@@ -158,30 +155,30 @@ class DataProxyWrapper : public cond::DataProxyWrapperBase {
...
 
 private:
   std::string m_source;
   edm::eventsetup::TypeTag m_type;
   std::vector<std::shared_ptr<cond::persistency::PayloadProxy<DataT>>> m_proxies;
-  std::vector<std::shared_ptr<DataProxy>> m_edmProxies;
+  std::vector<std::shared_ptr<ProductResolver>> m_esResolvers;
 };
 
 #endif /* CONDCORE_PLUGINSYSTEM_DATAPROXY_H */
```
Comment (francescobrivio) [inline CondCore/ESSources/interface/ProductResolver.h:184]: Just for consistency I would change this to:
```suggestion
#endif /* CONDCORE_PLUGINSYSTEM_PRODUCTRESOLVER_H */
```

**Comment by Dr15Jones** [comment]
please test

**Comment by wddgit** [comment]
isFullyResolved was deleted in the recently merged PR removing consumesMany...

**Comment by Dr15Jones** [comment]
please test

**Comment by francescobrivio** [comment]
+1

 - tests are clean
 - changes in `alca`/`db` part are according to description

**Comment by makortel** [comment]
+core

**Comment by mandrenguyen** [comment]
+1

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
merge

**Pull request merged**